src/Functions/XPath/XMLTagScanner.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "Types.h"
+
+#include <base/find_symbols.h>
+#include <Common/StringUtils.h>
+
+#include <string_view>
+
+namespace DB
+{
+
+struct XMLTagScanner : public ITagScanner
+{
+    /// @returns position of the symbol after a tag name or end if no tag is found
+    const char * scan(const char * begin, const char * end, TagPreview & result) override
```

Comment:
It could make sense to consider `Poco::XML::XMLStreamParser` or another ready XML parser to avoid reimplementing.
Or at least our parser tokenizer.